packages/multichain-api-middleware/src/handlers/wallet-createSession.ts

```diff
@@ -0,0 +1,250 @@
+ * @param hooks - The hooks object.
+ * @param hooks.listAccounts - The hook that returns an array of the wallet's evm accounts.
+ * @param hooks.findNetworkClientIdByChainId - The hook that returns the networkClientId for a chainId.
+ * @param hooks.requestPermissionsForOrigin - The hook that approves and grants requested permissions.
+ * @param hooks.getNonEvmSupportedMethods - The hook that returns the supported methods for a non EVM scope.
+ * @param hooks.isNonEvmScopeSupported - The hook that returns true if a non EVM scope is supported.
+ * @param hooks.getNonEvmAccountAddresses - The hook that returns a list of CaipAccountIds that are supported for a CaipChainId.
+ * @param hooks.trackSessionCreatedEvent - An optional hook for platform specific logic to run. Can be undefined.
+ * @returns A promise with wallet_createSession handler
+ */
+async function walletCreateSessionHandler(
+  req: JsonRpcRequest<Caip25Authorization> & { origin: string },
+  res: JsonRpcSuccess<{
+    sessionScopes: NormalizedScopesObject;
+    sessionProperties?: Record<string, Json>;
+  }>,
+  _next: JsonRpcEngineNextCallback,
+  end: JsonRpcEngineEndCallback,
+  hooks: {
+    listAccounts: () => { address: string }[];
+    findNetworkClientIdByChainId: NetworkController['findNetworkClientIdByChainId'];
+    requestPermissionsForOrigin: (
+      requestedPermissions: RequestedPermissions,
+      metadata?: Record<string, Json>,
+    ) => Promise<[GrantedPermissions]>;
+    getNonEvmSupportedMethods: (scope: CaipChainId) => string[];
+    isNonEvmScopeSupported: (scope: CaipChainId) => boolean;
+    getNonEvmAccountAddresses: (scope: CaipChainId) => CaipAccountId[];
+    trackSessionCreatedEvent?: () => void;
+  },
+) {
+  if (!isPlainObject(req.params)) {
+    return end(invalidParams({ data: { request: req } }));
+  }
+  const { requiredScopes, optionalScopes, sessionProperties } = req.params;
+
+  if (sessionProperties && Object.keys(sessionProperties).length === 0) {
+    return end(new JsonRpcError(5302, 'Invalid sessionProperties requested'));
+  }
+
+  const filteredSessionProperties = Object.fromEntries(
+    Object.entries(sessionProperties ?? {}).filter(([key]) =>
+      isKnownSessionPropertyValue(key),
+    ),
+  );
+
+  try {
+    const { normalizedRequiredScopes, normalizedOptionalScopes } =
+      validateAndNormalizeScopes(requiredScopes || {}, optionalScopes || {});
+
+    const requiredScopesWithSupportedMethodsAndNotifications =
+      getSupportedScopeObjects(normalizedRequiredScopes, {
+        getNonEvmSupportedMethods: hooks.getNonEvmSupportedMethods,
+      });
+    const optionalScopesWithSupportedMethodsAndNotifications =
+      getSupportedScopeObjects(normalizedOptionalScopes, {
+        getNonEvmSupportedMethods: hooks.getNonEvmSupportedMethods,
+      });
+
+    const networkClientExistsForChainId = (chainId: Hex) => {
+      try {
+        hooks.findNetworkClientIdByChainId(chainId);
+        return true;
+      } catch (err) {
```

Comment:
Nit: can you make this update
```suggestion
      } catch {
```

This way we remove

```
  "packages/multichain-api-middleware/src/handlers/wallet-createSession.ts": {
    "@typescript-eslint/no-unused-vars": 1
  },
```

Reply:
https://github.com/MetaMask/core/pull/5647/commits/81cf1dc0b0928378d56dc89a310e3ce2bc5b018a